### PR TITLE
move deletable logic down to the page tree component and refactor page tree

### PR DESCRIPTION
**Pull request by mattcasey**
Sorry for the refactor, the PageNavigation.tsx file was getting pretty big and i keep getting lost in it. i'll highlight the actual new logic below.

### CODE REVIEW

File: components/common/PageLayout/components/Sidebar.tsx [hunk 1]
```diff
@@ -198,48 +188,6 @@ export default function Sidebar ({ closeSidebar, favorites }: SidebarProps) {
     router.push(`/${newSpace.domain}`);
   }
 
-  const deletePage = useCallback(async (pageId: string) => {
```
Comment (mattcasey) [inline components/common/PageLayout/components/Sidebar.tsx:201]: i like that by moving the deletePage logic down to the popup menu, we no longer need to pull in the pages or focalboard boards in Sidebar!

File: components/common/PageLayout/components/PageNavigation/components/TreeNode.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
+
```
Comment (mattcasey) [inline components/common/PageLayout/components/PageNavigation/components/TreeNode.tsx:1]: no change here

File: components/common/PageLayout/components/PageNavigation/components/PageTreeItem.tsx [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+        anchorOrigin={anchorOrigin}
+        transformOrigin={transformOrigin}
+      >
+        {Boolean(anchorEl) && <PageActionsMenu closeMenu={closeMenu} pageId={pageId} pagePath={href} />}
+      </Menu>
+    </>
+  );
+});
+
+function PageActionsMenu ({ closeMenu, pageId, pagePath }: { closeMenu: () => void, pageId: string, pagePath: string }) {
```
Comment (mattcasey) [inline components/common/PageLayout/components/PageNavigation/components/PageTreeItem.tsx:367]: this is the menu now that appears when you click the "...". it opens very fast, at least for my 2-3 page workspace :). and all these hooks only get calculated when it opens

Comment (motechFR) [reply components/common/PageLayout/components/PageNavigation/components/PageTreeItem.tsx:367]: Really great optimisation!

**Review by motechFR — APPROVED** [review]
I like that you split up the files for easier navigation.

Would definitely add this code smell (files that are too large) to our code review standards when checking each others' PRs in future

File: components/common/PageLayout/components/PageNavigation/components/PageTreeItem.tsx (hunk 3 above)
Comment (mattcasey) [reply components/common/PageLayout/components/PageNavigation/components/PageTreeItem.tsx:367]: great suggestion!

**Pull request merged**